llama_stack/providers/remote/vector_io/chroma/chroma.py

```diff
@@ -193,18 +206,27 @@ async def query_chunks(
     ) -> QueryChunksResponse:
         index = await self._get_and_cache_vector_db_index(vector_db_id)
 
+        if index is None:
+            raise ValueError(f"Vector DB {vector_db_id} not found in Chroma")
+
         return await index.query_chunks(query, params)
 
     async def _get_and_cache_vector_db_index(self, vector_db_id: str) -> VectorDBWithIndex:
         if vector_db_id in self.cache:
             return self.cache[vector_db_id]
 
-        vector_db = await self.vector_db_store.get_vector_db(vector_db_id)
-        if not vector_db:
-            raise ValueError(f"Vector DB {vector_db_id} not found in Llama Stack")
-        collection = await maybe_await(self.client.get_collection(vector_db_id))
-        if not collection:
-            raise ValueError(f"Vector DB {vector_db_id} not found in Chroma")
-        index = VectorDBWithIndex(vector_db, ChromaIndex(self.client, collection), self.inference_api)
-        self.cache[vector_db_id] = index
-        return index
+        try:
+            collection = await maybe_await(self.client.get_collection(vector_db_id))
+            if not collection:
+                raise ValueError(f"Vector DB {vector_db_id} not found in Chroma")
+
+            vector_db = await self.vector_db_store.get_vector_db(vector_db_id)
+            if not vector_db:
+                raise ValueError(f"Vector DB {vector_db_id} not found in Llama Stack")
+
+            index = VectorDBWithIndex(vector_db, ChromaIndex(self.client, collection), self.inference_api)
+            self.cache[vector_db_id] = index
+            return index
+
+        except Exception as exc:
+            raise ValueError(f"Vector DB {vector_db_id} not found in Chroma") from exc
```

Comment:
this is really not correct -- you are eating all kinds of exceptions (clearly other errors can happen at runtime) and reporting them as "not found". why?

Reply:
Will revise to provide more explicit. 👍

Reply:
Will revise to provide more explicit. 👍

Reply:
i reverted the change and instead updated the unit test to capture the chroma exception that's triggered from `test_query_unregistered_raises`.